mindmeld/components/dialogue.py

```diff
@@ -914,15 +914,26 @@ def _prompt_slot(self, responder, nlr):
         self._retry_attempts = 0
         self._prompt_turn = False
 
-    def _retry_logic(self, responder, nlr):
+    def _retry_logic(self, request, responder, nlr):
         if self._retry_attempts < self._max_retries:
             self._retry_attempts += 1
             responder.reply(nlr)
         else:
             # max attempts exceeded, reset counter, exit auto_fill.
             self._retry_attempts = 0
             self._exit_flow(responder)
-            self._app.app_manager.dialogue_manager.reprocess()
+
+            processed_query = self._app.app_manager.nlp.process(query_text=request.text)
```

Comment:
Should `request` already encapsulate all of the fields?

Reply:
No it doesn't because the allowed intents were set at the turn before this, hence the request before this step has fixed intent and param definition. nlp.process here is applied after removing the allowed_intents.

Reply:
where is that removal being done?

Reply:
In ``self._exit_flow``, line 924. That function resets both the ``target_dialogue_state`` and ``allowed_intents``

Reply:
I see got it, can you add a comment here about why we need to reprocess the NLP component